File: CodeWars/6kyu/sum-of-digits-slash-digital-root/main.js
// My Method 1~ 🦉
function digital_root(n) {
    const sum_numbers = (number_string) => {
        let sum = 0;
        for (let i = 0; i < number_string.length; i++) {
            sum += Number(number_string[i]);
        }
        return sum;
    };

    result = sum_numbers(String(n));

    while (result > 9) {
        result = sum_numbers(String(result));
    }

    return result;
}

const test_case = digital_root(493193);
console.log(test_case);

File: CodeWars/6kyu/sum-of-digits-slash-digital-root/main.py
# My Method 1~ 🦉
def digital_root(n):
    def sum_elements(number):
        return sum([int(num) for num in str(number)])

    sum_final = sum_elements(n)
    while sum_final > 9:
        sum_final = sum_elements(sum_final)

    return sum_final


test_case = digital_root(493193)
print(test_case)
